File: src/McCardz.Domain/Models/Answer.cs
﻿using McCardz.Domain.Repositories;

namespace McCardz.Domain.Models;

public class Answer : IEntity
{
    public string Text { get; set; } = string.Empty;

    public bool IsCorrect { get; set; }

    public bool IsAiGenerated { get; set; }

    public int QuestionId { get; set; }
    public int Id { get; set; }
}

File: src/McCardz.Web/ClientApp/src/models/Answer.ts
export default class Answer {
    id: string = '';
    questionId: string = '';
    text: string = '';
    isCorrect: boolean = false;
    isAiGenerated: boolean = false;
    
    constructor(id: string, questionId: string, text: string, isCorrect: boolean, isAiGenerated: boolean) {
        this.id = id;
        this.questionId = questionId;
        this.text = text;
        this.isCorrect = isCorrect;
        this.isAiGenerated = isAiGenerated;
    }
}